base/src/main/java/com/smartdevicelink/managers/screen/choiceset/BaseChoiceSetManager.java

```diff
@@ -452,17 +453,68 @@ public void dismissKeyboard(@NonNull Integer cancelID) {
      * @param keyboardConfiguration - the custom keyboard configuration to be used when the keyboard is displayed
      */
     public void setKeyboardConfiguration(@Nullable KeyboardProperties keyboardConfiguration) {
+        processKeyboardConfiguration(keyboardConfiguration);
+    }
+
+    private KeyboardProperties processKeyboardConfiguration(@Nullable KeyboardProperties keyboardConfiguration) {
```

Comment:
Can we replace `processKeyboardConfiguration` method with this method. THe suggested method is much simpler and aligns with iOS implementation.

```
// Takes a keyboard configuration (SDLKeyboardProperties) and creates a valid version of it, if possible, based on this object's internal keyboardCapabilities
private KeyboardProperties createValidKeyboardConfigurationBasedOnKeyboardCapabilitiesFromConfiguration(@Nullable KeyboardProperties keyboardConfiguration) {
    KeyboardCapabilities keyboardCapabilities = defaultMainWindowCapability != null ? defaultMainWindowCapability.getKeyboardCapabilities() : null;

    // If there are no keyboard capabilities, if there is no passed keyboard configuration, or if there is no layout to the passed keyboard configuration, just pass back the passed in configuration
    if (keyboardCapabilities == null || keyboardConfiguration == null || keyboardConfiguration.getKeyboardLayout() == null) {
        return keyboardConfiguration;
    }

    KeyboardLayoutCapability selectedLayoutCapability = null;
    for (KeyboardLayoutCapability layoutCapability : keyboardCapabilities.getSupportedKeyboards()) {
        if (layoutCapability.getKeyboardLayout().equals(keyboardConfiguration.getKeyboardLayout())) {
            selectedLayoutCapability = layoutCapability;
            break;
        }
    }

    if (selectedLayoutCapability == null) {
        DebugTool.logError(TAG, String.format("Configured keyboard layout is not supported: %s", keyboardConfiguration.getKeyboardLayout()));
        return null;
    }

    KeyboardProperties modifiedKeyboardConfiguration = (KeyboardProperties) keyboardConfiguration.clone();

    if (keyboardConfiguration.getCustomKeys() == null || keyboardConfiguration.getCustomKeys().isEmpty()) {
        modifiedKeyboardConfiguration.setCustomKeys(null);
    } else {
        // If there are more custom keys than are allowed for the selected keyboard layout, we need to trim the number of keys to only use the first n number of custom keys, where n is the number of allowed custom keys for that layout.
        int numConfigurableKeys = selectedLayoutCapability.getNumConfigurableKeys();
        if (keyboardConfiguration.getCustomKeys().size() > numConfigurableKeys) {
            modifiedKeyboardConfiguration.setCustomKeys(keyboardConfiguration.getCustomKeys().subList(0, numConfigurableKeys));
            DebugTool.logWarning(TAG, String.format(Locale.US, "%d custom keys set, but the selected layout: %s only supports %d. Dropping the rest.", keyboardConfiguration.getCustomKeys().size(), keyboardConfiguration.getKeyboardLayout(), numConfigurableKeys));
        }
    }

    // If the keyboard does not support masking input characters, we will remove it from the keyboard configuration
    if (!keyboardCapabilities.getMaskInputCharactersSupported()) {
        modifiedKeyboardConfiguration.setMaskInputCharacters(null);
    }

    return modifiedKeyboardConfiguration;
}
```

Reply:
Also please add unit tests to test the newly added manager code.